include/megaapi.h

```diff
@@ -12529,6 +12684,103 @@ class MegaApi
          */
         bool createAvatar(const char *imagePath, const char *dstPath);
 
+        /**
+         * @brief Initial step to upload a photo/video via iOS low-power background upload feature
+         *
+         * Call ensureMediaInfo() first in order prepare the library to attach file attributes
+         * that enable videos to be identified and played in the web browser.
+         *
+         * @return A pointer to an object that keeps some needed state through the process of
+         *         uploading a media file via iOS low power background uploads (or similar).
+         *         Caller takes ownership of the object.
+         */
+        MegaBackgroundMediaUpload* backgroundMediaUploadNew();
+
+        /**
+         * @brief Get back the needed MegaBackgroundMediaUpload after the iOS app exited and restarted
+         *
+         * In case the iOS app exits while a background upload is going on, and the app is started again
+         * to complete the operation.  Call this version to recreate the MegaBackgroundMediaUpload object
+         * needed for a call to backgroundMediaUploadComplete.   The object must have been serialised
+         * before the app was unloaded.
+         *
+         * @return A pointer to an object that keeps some needed state through the process of
+         *         uploading a media file via iOS low power background uploads (or similar).
+         *         Caller takes ownership of the object.
+         */
+        MegaBackgroundMediaUpload* backgroundMediaUploadResume(const std::string* serialised);
```

Comment:
I started to make this change, and discovered why I did it that way - MegaBackgroundMediaUpload needs to store a MegaApiImpl* for future use, and immediate use.   That pointer is only available via the MegaApi.   It's not the first class to store one of those pointers, but it's the first to do that and support serialization.   What's the best way to resolve that?   thanks!

Reply:
The best way to solve it would be to add the pointer to `MegaApi` to the params, to both of them:
```
static MegaBackgroundMediaUpload* MegaBackgroundMediaUpload::unserialize(const char* data, MegaApi* api);
```
```
static MegaBackgroundMediaUpload* MegaBackgroundMediaUpload::createInstance(MegaApi *api)
```
You can see the example above for the `createInstance()`, for which I provided a possible implementation it my comment.